lib/src/platform_client.dart

```diff
@@ -837,6 +858,32 @@ class PlatformClient {
     await _httpPost('/api/smartapp/token', body);
   }
 
+  /// Requests Platform if there is an available Site Offer at the current position.
+  ///
+  /// Returns null if there is no Site Offer available or an `AccessOfferDetails` of the available offer if within
+  /// geofence.
+  Future<AccessOfferDetails?> inquireForGPSActivatedOffer(Position position) async {
+    _verifyIsLoggedIn();
+
+    if (null != _lastPositionUpdate && _lastPositionUpdate!.timeSinceInMs() < 1000) {
+      // Same throttling delay as `KurrentoRoom.updateLocation`.
+      return _lastAccessOfferUpdate;
+    }
```

Comment:
I think this throttling would make more sense if the method was named `updateLocation`. If it's an `inquire` method, it doesn't feel right to return stale/incorrect data.

Reply:
I'll change the name.

Reply:
I have added a nice comment explaining why we do the throttling. To me, `updateLocation` doesn't sound right because it doesn't update anything on the server side.